woocommerce/woocommerce-ios
# [Shipping labels] Get account settings and use in UI custom packages

Closes  #14484



## Description


- Loading store options from backend so it can be used when creating custom packages (for **weight** and **dimension** units)
- I have a work in progress branch where I have been working on logic for persisting the store options in Core Data but I am wondering if we should do it or not? The main thing I am thinking about is whether we always need "fresh" store options or not, if we need fresh ones then the persisting them and loading them does not make much sense since we will need to get the fresh ones. @rachelmcr what do you think?

## Testing information


- Install and set up the Woo Shipping extension on your store.
- Build and run the app with the revampedShippingLabelCreation feature flag enabled.
- Create an order with the processing status and at least one physical product.
- In the order details, select "Create Shipping Label."
- Store options will start fetching from backend in the background.
- Tap "Select a Package" button
- If store options are already loaded you will see the UI, if not you will see the loading indicator until the loading finishes. If it fails you will see a reload button that triggers the loading again.
- Check that the fields have proper weight and dimensions units (turn template switch ON to see weight unit).

## Screenshots


**Flow when store options are not loaded yet and users taps fast thru UI and gets to UI part that needs it:**

https://github.com/user-attachments/assets/0439ca2d-142e-4bdc-bd26-a6ac6c3c424a

**Store options already loaded and UI has everything needed:**

https://github.com/user-attachments/assets/4a3613b1-dccd-4a73-858a-3562c1f78318

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
